### DISCUSSION TITLE
<userdoc> doesn't work as expected.

### DISCUSSION THREAD

**Discussion by Bodyclock-Games** [stride3d/stride#1925] [Q&A] [answered]
I have the following abstract class, part of a hierarchy of classes for the editor. This is a way down the chain of inheritance. I wanted to add some editor documentation so I added the Priority <userdoc> tag (which works), copied directly from the ScriptComponent code, but it doesn't show up. I tried inheriting from entity component but no joy. Am I missing something?

```
    public abstract class DomainWarpFractal
    {
        #region PUBLIC PROPERTIES
        [DataMember(510)]
        [Display("Octaves")]
        public int Octaves { get; set; } = 2;

        [DataMember(520)]
        [Display("Lacunarity")]
        public float Lacunarity { get; set; } = 2.0f;

        /// <summary>
        /// The priority this script will be scheduled with (compared to other scripts).
        /// </summary>
        /// <userdoc>The execution priority for this script. It applies to async, sync and startup scripts. Lower values mean earlier execution.</userdoc>
        [DataMember(530)]
        [Display("Gain")]
        public float Gain { get; set; } = 2.0f;
        #endregion

        #region STATIC

        #region PUBLIC PROPERTIES
        public static Noise.FractalType FractalType { get; protected set; }
        #endregion

        #endregion
    }
```

**Comment by manio143** [comment, marked as answer]
I believe you need to add `GenerateDocumentationFile` property to your csproj and set it to true.

**Reply by Eideren** [reply]
More info in the manual over there https://doc.stride3d.net/latest/en/manual/scripts/public-properties-and-fields.html?q=userdoc#adding-property-descriptions

**Reply by Bodyclock-Games** [reply]
Thanks
